pkg/apis/configuration/validation/validation.go

```diff
@@ -50,12 +50,53 @@ func validateHost(host string, fieldPath *field.Path) field.ErrorList {
 }
 
 func validateTLS(tls *v1alpha1.TLS, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
 	if tls == nil {
 		// valid case - tls is not defined
-		return field.ErrorList{}
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validateSecretName(tls.Secret, fieldPath.Child("secret"))...)
+
+	allErrs = append(allErrs, validateTLSRedirect(tls.Redirect, fieldPath.Child("redirect"))...)
+
+	return allErrs
+}
+
+func validateTLSRedirect(redirect *v1alpha1.TLSRedirect, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if redirect == nil {
+		return allErrs
 	}
 
-	return validateSecretName(tls.Secret, fieldPath.Child("secret"))
+	if redirect.Code != nil {
+		allErrs = append(allErrs, validateTLSRedirectStatusCode(*redirect.Code, fieldPath.Child("code"))...)
+	}
+
+	if redirect.BasedOn != "scheme" && redirect.BasedOn != "x-forwarded-proto" {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("basedOn"), redirect.BasedOn, "accepted values are 'scheme' or 'x-forwarded-proto'"))
+	}
+
+	return allErrs
+}
+
+var validTLSRedirectStatusCodes = map[int]bool{
+	301: true,
+	302: true,
+	303: true,
```

Comment:
Please remove `303` and add `308` and make that change reflected in the other places like the docs.